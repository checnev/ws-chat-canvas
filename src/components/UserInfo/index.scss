.userInfo {
  display: flex;
  align-items: center;

  &__avatar {
    display: flex;
    min-width: 32px;
    height: 32px;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    border: 1px solid rgba(0, 0, 0, 0.1);
    color: rgb(0, 0, 0);
    font-weight: bold;
  }
  
  &__username {
    margin-left: 10px;
  }
}
