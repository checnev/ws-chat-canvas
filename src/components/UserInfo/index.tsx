import React from 'react';
import cn from 'utils/cn';
import clsx from 'clsx';
import './index.scss';
import { IUser } from 'lib/user/User';

interface UserInfoProps {
  user: IUser;
  className?: string;
}

class UserInfo extends React.Component<UserInfoProps> {

  render() {
    const userInfo = cn('userInfo');

    return (
      <div
        className={clsx(this.props.className, userInfo())}
      >
        <div
          className={userInfo('avatar')}
          style={{ backgroundColor: this.props.user.color }}
        >
          {this.props.user.name[0].toUpperCase()}
        </div>
        <span className={userInfo('username')}>{this.props.user.name}</span>
      </div>
    );
  }
}

export default UserInfo;
