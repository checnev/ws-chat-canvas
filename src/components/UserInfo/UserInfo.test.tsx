import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import UserInfo from 'components/UserInfo';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render UserInfo', () => {
  act(() => {
    render(<UserInfo user={{ name: 'Naruto', color: 'green' }} />, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
