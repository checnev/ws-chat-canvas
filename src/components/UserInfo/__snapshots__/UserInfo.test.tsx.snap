// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserInfo 1`] = `
"<div class=\\"userInfo\\">
  <div class=\\"userInfo__avatar\\" style=\\"background-color: green;\\">N</div><span class=\\"userInfo__username\\">Naruto</span>
</div>"
`;
